### PR TITLE
CI/CD setup

### PR DESCRIPTION
**Pull request by coryodaniel**
* [Refactor] examples into test fixtures
* [Added] Makefile from cookiecutter template

### CODE REVIEW

File: test/boilerplate/boilerplate.Dockerfile.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2018 Google LLC
```
Comment (Jberlinsky) [inline test/boilerplate/boilerplate.Dockerfile.txt]: See above about updating to 2019. Not going to duplicate further :)

File: test/fixtures/advanced/example.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+module "example" {
+  source = "../../../examples/advanced"
+
+  project_id       = "${var.project_id}"
+  credentials_path = "${local.credentials_path}"
+  region           = "${var.region}"
+  name_suffix      = "-${random_string.suffix.result}"
```
Comment (Jberlinsky) [inline test/fixtures/advanced/example.tf]: See above about `var.names`. Not going to duplicate further.

File: test/fixtures/config.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (Jberlinsky) [inline test/fixtures/config.sh]: This file shouldn't be necessary

File: test/integration/advanced/controls/gcloud.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    let!(:data) do
+      if subject.exit_status == 0
+        JSON.parse(subject.stdout)
+      else
+        {}
+      end
+    end
+
+    describe "nat" do
+      it "did not generate a NAT name" do
```
Comment (Jberlinsky) [inline test/integration/advanced/controls/gcloud.rb]: Can we instead assert that it has the NAT name we _expect_?

File: test/integration/basic/controls/gcloud.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    let!(:data) do
+      if subject.exit_status == 0
+        JSON.parse(subject.stdout)
+      else
+        {}
+      end
+    end
+
+    describe "nat" do
+      it "generated a NAT name" do
```
Comment (Jberlinsky) [inline test/integration/basic/controls/gcloud.rb]: See above about asserting positively

File: test/make.sh [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  echo "Running hadolint on Dockerfiles"
+  find . -name "Dockerfile" -exec hadolint {} \;
+}
+
+# This function runs 'terraform validate' against all
+# files ending in '.tf'
+function check_terraform() {
+  echo "Running terraform validate"
+  #shellcheck disable=SC2156
+  find . -name "*.tf" -exec bash -c 'terraform validate --check-variables=false $(dirname "{}")' \;
```
Comment (Jberlinsky) [inline test/make.sh]: This looks like it's missing a few `-not path`s from the module template. Does `make` succeed?

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+variable "project_id" {
+  description = "The project ID to deploy to"
+}
+
+variable "region" {
+  description = "The region to deploy to"
+}
+
+variable "icmp_idle_timeout_sec" {
+  description = "(Optional) Timeout (in seconds) for ICMP connections. Defaults to 30s if not set. Changing this forces a new NAT to be created."
```
Comment (Jberlinsky) [inline variables.tf]: Not sure how I feel about `(Optional)` in the description here. Doesn't having a `default` imply this?

File: test/integration/basic/controls/gcloud.rb (hunk 5 above)
Comment (coryodaniel) [reply test/integration/basic/controls/gcloud.rb]: In this example its testing that the module itself (not the fixture or example) generated a name.

**Pull request merged**